File: Code Interviews Solution/链表中倒数第k个结点.cpp
/*
struct ListNode {
	int val;
	struct ListNode *next;
	ListNode(int x) :
			val(x), next(NULL) {
	}
};*/
class Solution {
public:
    ListNode* FindKthToTail(ListNode* pListHead, unsigned int k) {
    	if(pListHead == NULL)
            return NULL;
        int len = 0;
        ListNode* tmp = pListHead;
        while(tmp != NULL) {
            tmp = tmp->next;
            len++;
        }
        if(k > len || k == 0)
            return NULL;
        int index = 0;
        int pivot = len - k;
        tmp = pListHead;
        while(index != pivot) {
            tmp = tmp->next;
            index++;
        }
        return tmp;
    }
};

File: Code Interviews Solution/链表中倒数第k个结点.java
/*
public class ListNode {
    int val;
    ListNode next = null;

    ListNode(int val) {
        this.val = val;
    }
}*/
public class Solution {
    public ListNode FindKthToTail(ListNode head,int k) {
		if(head == null || k <= 0)
            return null;
        ListNode tmp = head;
        int len = 0;
        while(tmp != null) {
            tmp = tmp.next;
            len++;
        }
        if ( k > len)
            return null;
        int index = 0;
        int pivot = len - k;
        tmp = head;
        while(index != pivot) {
            tmp = tmp.next;
            index++;
        }
        return tmp;
    }
}